packages/tools/src/stateManagement/segmentation/SegmentationStateManager.ts

```diff
@@ -497,225 +566,153 @@ export default class SegmentationStateManager {
+  ): SegmentationRepresentation[] | [] {
+    const segmentationRepresentations = this.state.viewports[viewportId];
+
+    if (!segmentationRepresentations) {
       return;
     }
 
-    _config.allSegments = config;
-  }
-
-  /**
-   * Sets the configuration for per-segment settings of a segmentation representation.
-   *
-   * @param segmentationRepresentationUID - The unique identifier of the segmentation representation.
-   * @param config - The configuration for per-segment settings.
-   */
-  setPerSegmentConfig(
-    segmentationRepresentationUID: string,
-    config: RepresentationConfig
-  ): void {
-    const _config = this._getRepresentationConfig(
-      segmentationRepresentationUID
-    );
-
-    if (!_config) {
-      return;
+    // If no specifier is provided, return all entries
+    if (!specifier.type && !specifier.segmentationId) {
+      return segmentationRepresentations;
     }
 
-    _config.perSegment = config;
+    return segmentationRepresentations.filter((representation) => {
+      const typeMatch = specifier.type
+        ? representation.type === specifier.type
+        : true;
+      const idMatch = specifier.segmentationId
+        ? representation.segmentationId === specifier.segmentationId
+        : true;
+      return typeMatch && idMatch;
+    });
   }
 
   /**
-   * Returns the visibility of a segmentation representation in a specific viewport.
+   * Retrieves the visibility of a segmentation representation in a specific viewport.
    * @param viewportId - The ID of the viewport.
-   * @param segmentationRepresentationUID - The UID of the segmentation representation.
+   * @param segmentationId - The ID of the segmentation.
+   * @param type - The type of the segmentation representation.
    * @returns The visibility of the segmentation representation in the viewport.
    */
   getSegmentationRepresentationVisibility(
     viewportId: string,
-    segmentationRepresentationUID: string
+    segmentationId: string,
+    type: SegmentationRepresentations
   ): boolean {
-    const viewport = this.state.viewports[viewportId];
-    return viewport && viewport[segmentationRepresentationUID]?.visible;
+    const viewportRepresentations = this.getSegmentationRepresentations(
+      viewportId,
+      {
+        segmentationId,
+        type,
+      }
+    );
+
+    return viewportRepresentations?.[0]?.visible;
   }
 
   /**
    * Sets the visibility of a segmentation representation in a specific viewport.
    * @param viewportId - The ID of the viewport.
-   * @param segmentationRepresentationUID - The UID of the segmentation representation.
+   * @param segmentationId - The ID of the segmentation.
+   * @param type - The type of the segmentation representation.
    * @param visible - The visibility to set for the segmentation representation in the viewport.
    */
   setSegmentationRepresentationVisibility(
     viewportId: string,
-    segmentationRepresentationUID: string,
+    segmentationId: string,
```

Comment:
Allow either of thse two to be not provided., as then one could set set invisble all contour types.